src/Findologic/Api/FindologicSearchService.php

```diff
@@ -213,4 +203,52 @@ protected function isCategoryPage(NavigationRequestHandler $handler, ProductList
 
         return !empty($isCategoryPage);
     }
+
+    public function doFilter(ProductListingCriteriaEvent $event): void
+    {
+        $limit = self::FILTER_REQUEST_LIMIT;
+        $handler = $this->buildNavigationRequestHandler();
+        if (!$this->isCategoryPage($handler, $event)) {
+            $handler = $this->buildSearchRequestHandler();
+        }
+
+        $this->handleFilters($event, $handler);
+        $this->handleSelectableFilters($event, $handler, $limit);
+    }
+
+    protected function handleSelectableFilters(
+        ProductListingCriteriaEvent $event,
+        SearchNavigationRequestHandler $requestHandler,
+        ?int $limit
+    ): void {
+        $response = $requestHandler->doRequest($event, $limit, true);
+        $filtersWithSmartSuggestBlocks = $this->parseFiltersFromResponse($response, $event);
+
+        $event->getCriteria()->addExtension('flAvailableFilters', $filtersWithSmartSuggestBlocks);
+    }
+
+    /**
+     * @param Response $response
+     * @param ProductListingCriteriaEvent $event
+     *
+     * @return mixed
```

Comment:
Add a strict type hint and add it as method return type.

Reply:
I also saw that this wasn't explicitly added to the `ResponseParser` class, therefore also add `FiltersExtension` as return type for this method in the response parser.